go-app-ussd_popi_user_data.js

```diff
@@ -185,19 +185,27 @@ go.app = function() {
                 
                 // display in previously chosen language
                 self.im.user.set_lang(self.im.user.answers.state_language);
-                
                 return sbm
                 // check that user is registered on momconnect
+                                 
                 .check_identity_subscribed(self.im.user.answers.operator.id, "momconnect")
                 .then(function(identity_subscribed_to_momconnect) {
                     if (identity_subscribed_to_momconnect) {
-                        var mset = sbm.list_active_subscriptions();
-                        var allmset = '';
-                        for(var i = 0; i < mset.length; i++){
-                            allmset += sbm.get_messageset(mset[i]);
-                        }
-                        self.im.user.set_answer("message_sets",allmset);
-                        return self.states.create('state_all_options_view');
+                        
+                        return sbm
+                        .list_active_subscriptions(self.im.user.answers.operator.id)
+                        .then(function(active_subscriptions){
```

Comment:
I think you'll need to loop through the active subscriptions here. Right now you'll only get the message set for one of their subscriptions. It's unlikely but possible that a user might have multiple subscriptions.